url063b9533f6_karlney_tetris_ai_tgz-pJ8-karlney_tetris_cmd_TetrisGameRunnerJ8/src/karlney/tetris/swing/TetrisGameSwing.java

```diff
@@ -0,0 +1,405 @@
+            changeLevelCmd,
+            showHighscoreCmd,
+            nextLvlCmd,
+            prevLvlCmd;
+
+
+    public TetrisGame game;
+    public PieceGenerator generator = new RandomPieceGenerator();
+
+
+    /**
+     * Tis method starts a new game
+     */
+    private void newGame() {
+        int level = DEFAULT_START_LEVEL;
+        if (game!=null){
+            game.stop();
+            level = game.getLevel();
+        }
+
+        Board board = BoardBuilder.createBoard(10, 20, new int[][]{
+                {1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0},
+                {1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0},
+                {1, 1, 0, 1, 0, 1, 1, 1, 1, 0, 0, 0},
+                {1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0, 0},
+                {1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
+        });
+
+        //Player player = new Player(new Board(),generator,level);
+        Player player = AIPlayerBuilder.getInstantMoveOnePieceAIPlayer(board, generator, new ElAshiTetrisBoardEvaluator(), level, 0);
+
+        game = new TetrisGame(level,Arrays.asList(player));
+        game.start();
+    }
+
+    public TetrisGameSwing(){
+        initGraphics();
+        new Thread(this).start();
+        newGame();
+    }
+
+    private void initGraphics() {
+        jf.setResizable(false);
+        jf.setSize(new Dimension(XSIZE,YSIZE));
+        jf.setLocation(LOCATION_X,LOCATION_Y);
+        jf.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        jf.addKeyListener(this);
+        Container c=jf.getContentPane();
+        setMinimumSize(new Dimension(XSIZE,YSIZE));
+        c.add(this);
+        buildMenu(jf);
+        jf.setVisible(true);
+    }
+
+
+    public void buildMenu(JFrame jf){
+        jf.setJMenuBar(new JMenuBar());
+        jf.getJMenuBar().setBorderPainted(false);
+
+        JMenu	m1= new JMenu("Game");
+        m1.setBackground(new Color(150,150,250));
+
+        newGameCmd=m1.add(new JMenuItem("New Game		(Ctrl+N)"));
+        newGameCmd.addActionListener(this);
+
+        pauseCmd=m1.add(new JMenuItem("Pause/Unpause	(P)"));
+        pauseCmd.addActionListener(this);
+
+        prevLvlCmd=m1.add(new JMenuItem("Decrease	 Level	  (1)"));
+        prevLvlCmd.addActionListener(this);
+
+        nextLvlCmd=m1.add(new JMenuItem("Increase	 Level	  (2)"));
+        nextLvlCmd.addActionListener(this);
+
+        exitCmd=m1.add(new JMenuItem("Exit	  (Ctrl+Q)"));
+        exitCmd.addActionListener(this);
+
+        jf.getJMenuBar().add(m1);
+        jf.getJMenuBar().setBackground(new Color(0,0,0));
+
+        newGame();
+    }
+
+
+    /**
+     * This methods shuts down the JVM
+     */
+    public void	quitGame(){
+        System.exit(0);
+    }
+
+    // User	Input
+    public void	keyTyped(KeyEvent	e){}
+    public void	keyReleased(KeyEvent	e){}
+
+    /**
+     * This method is called when a key is pressed
+     *
+     * @param e2 key pressed event
+     */
+    public void	keyPressed(KeyEvent e2){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.